manager.py

```diff
@@ -123,16 +123,27 @@ def stop(self):
     def _run_config_update(self):
         # Callback function to run update at the end of each polling interval
 
-        # TODO: In NIO-1142, this should first get the newest config version 
-        # from the product api before triggering an update
-        url = "{}/{}/versions/{}".format(self.config_api_url_prefix,
-                                         self.config_id,
-                                         self.config_version_id)
-        result = self.update_configuration(url)
-        self.logger.info("Configuration was updated, {}".format(result))
-
-    def update_configuration(self, url):
-        configuration = self._api_proxy.load_configuration(url, self.api_key) or {}
+        # Poll the product api for the latest config version id
+        latest_version_id = \
+            self._api_proxy.get_version(self.config_api_url_prefix,
+                                        self.config_id,
+                                        self.api_key)
+        config_version_id = \
+            latest_version_id.get("instance_configuration_version_id")
+        # Update instance with new config version id
+        if config_version_id != self.config_version_id:
+            self.config_version_id = config_version_id
+            result = self.update_configuration(self.config_api_url_prefix,
+                                               self.config_id,
+                                               self.config_version_id)
+            self.logger.info("Configuration was updated, {}".format(result))
+
+    def update_configuration(self, url_prefix, config_id, config_version_id):
+        configuration = \
+            self._api_proxy.load_configuration(url_prefix,
```

Comment:
This `or {}` scares me without any additional logging or anything. When do we expect the `load_configuration` call to return a `None` or anything falsey? Assuming it's an empty configuration is also scary, does that mean it will delete all my blocks and services if there is some kind of error? It seems like an error condition to me so I would suggest replacing with something like this:
```python
configuration = self._api_proxy.load_configuration(...)
if not configuration:
    self.logger.error("No configuration found, aborting update")
    return
```

Reply:
Yeah thats true. The api would actually always return a JSON response containing at the very least `message`, `status`, `uuid`, and `version_num` with a potential empty `configuration_data` field. I'm good with logging an error if we don't get a response

Reply:
Scratch that. We do have an error message that gets logged on line 147 if the response is missing `configuration_data` which aborts the update. I'll remove the `{}` though as I don't think it's needed